pkg/grizzly/grafana.go

```diff
@@ -22,12 +22,22 @@ type Folder struct {
 }
 
 // Board enscapsulates a dashboard for upload to Grafana API
-type Board struct {
-	Dashboard map[string]interface{} `json:"dashboard"`
-	Overwrite bool                   `json:"overwrite"`
+type APIBoard struct {
+	Dashboard Dashboard `json:"dashboard"`
+	Overwrite bool      `json:"overwrite"`
+}
+
+func (a APIBoard) String() string {
+	data, err := json.Marshal(a)
+	if err != nil {
+		panic(err)
+	}
+	return string(data)
 }
 
-func (b Board) String() string {
+type Dashboard map[string]interface{}
+
+func (b Dashboard) String() string {
```

Comment:
This doesn't seem required for the fix, please break a refactoring apart from the fix so the reviewer can see what is going on.

Reply:
It actually is kind of.

The problem was the fact that the old `Board` type was not able to reflect both, the api representation and the plain one used by Jsonnet.

Unmarshalling both into the same data structure rendered one or the other unusable. Instead of hacking around this, I decided to separate the types, so it works properly.
As a side-effect, diff started to work again